### PR TITLE
License-Scanning feature  for IETS3.Opensource

### PR DESCRIPTION
**Pull request by arimer**
Porting License-Scanning feature to older maintenance branches.

### CODE REVIEW

**Comment by arimer** [comment]
You can see the result of this scaning here: https://mps.builds.itemis.cloud/repository/download/Iets3_BuildOrgIets3core/364486:id/sbom.json

File: gradle.properties [hunk 1]
```diff
@@ -1,3 +1,7 @@
+group=org.iets3
+# Project name cannot be defined here, see settings.gradle#rootProject.name instead
+#name=opensource
```
Comment (sergej-koscejev) [inline gradle.properties:3]: A bit unusual place for the group and name, I would put the group in build.gradle and the project name in settings.gradle, as the comment says, and remove these changes.

File: build.gradle [hunk 2]
```diff
@@ -467,3 +490,18 @@ githubRelease {
...
+    // The file name for the generated SBOMs (before the file format suffix)
+    outputName = "sbom"
+    // The file format generated, can be xml, json or all for generating both
+    outputFormat = "json"
+    // Don't include license texts in generated SBOMs
+    includeLicenseText = false
+}
+
+assemble {
+    dependsOn << cyclonedxBom
```
Comment (sergej-koscejev) [inline build.gradle:506]: dependsOn('cyclonedxBom') is more modern Gradle style

Comment (sergej-koscejev) [inline build.gradle:506]:
```suggestion
    dependsOn('cyclonedxBom')
```

File: build.gradle [hunk 3]
```diff
@@ -327,6 +329,27 @@ static def addDependency(Object pom, Configuration config) {
 
 }
 
+ext.additionalPomInfo = {
```
Comment (sergej-koscejev) [inline build.gradle:332]: Is this additional info used anywhere? I don't see it.

File: build.gradle [hunk 4]
```diff
@@ -18,6 +18,8 @@ plugins {
     id 'base'
     id 'maven-publish'
     id "com.github.breadmoirai.github-release" version "2.4.1"
+    id "org.cyclonedx.bom" version "1.8.2"
+    id "com.dorongold.task-tree" version "4.0.0"
```
Comment (sergej-koscejev) [inline build.gradle:22]: task-tree was probably only added for debugging, right? We can remove it, I think.

**Comment by sergej-koscejev** [comment]
Overall this seems to work but I think the additional POM info is not used, this should be fixed. If I'm wrong, an explanatory comment in the code would be appreciated.

File: build.gradle (hunk 3 above)
Comment (sergej-koscejev) [reply build.gradle:332]: https://artifacts.itemis.cloud/repository/maven-mps-snapshots/org/iets3/opensource/feature-licscan20222.2022.2.7467.a20d900-SNAPSHOT/opensource-feature-licscan20222.2022.2.7467.a20d900-20240709.103218-1.pom doesn't have it.

File: gradle.properties (hunk 1 above)
Comment (stephaneberle9) [reply gradle.properties:3]: Refactored as suggested

File: build.gradle (hunk 4 above)
Comment (stephaneberle9) [reply build.gradle:22]: Have removed task-tree again

**Comment by sergej-koscejev** [comment]
@stephaneberle9 I think you forgot to push.

**Comment by stephaneberle9** [comment]
> @stephaneberle9 I think you forgot to push.
I did, but to the wrong branch (`feature/licscan` rather than `feature/licscan20222`). Fixed that now.

**Review by sergej-koscejev — APPROVED** [review]

**Comment by arimer** [comment]
@sergej-koscejev Thanks for the review 👍🏼

**Pull request merged**